src/osm_lua_processing.cpp

```diff
@@ -338,28 +338,31 @@ void OsmLuaProcessing::LayerAsCentroid(const string &layerName) {
 }
 
 // Set attributes in a vector tile's Attributes table
-void OsmLuaProcessing::Attribute(const string &key, const string &val) {
+void OsmLuaProcessing::Attribute(const string &key, const string &val) { AttributeWithMinZoom(key,val,0); }
+void OsmLuaProcessing::AttributeWithMinZoom(const string &key, const string &val, const char minzoom) {
 	if (val.size()==0) { return; }		// don't set empty strings
 	if (outputs.size()==0) { cerr << "Can't add Attribute " << key << " if no Layer set" << endl; return; }
 	vector_tile::Tile_Value v;
 	v.set_string_value(val);
-	outputs.back().second.push_back(attributeStore.store_key_value(key, v));
+	outputs.back().second.push_back(attributeStore.store_key_value(key, v, minzoom));
 	setVectorLayerMetadata(outputs.back().first->layer, key, 0);
 }
 
-void OsmLuaProcessing::AttributeNumeric(const string &key, const float val) {
+void OsmLuaProcessing::AttributeNumeric(const string &key, const float val) { AttributeNumericWithMinZoom(key,val,0); }
+void OsmLuaProcessing::AttributeNumericWithMinZoom(const string &key, const float val, const char minzoom) {
 	if (outputs.size()==0) { cerr << "Can't add Attribute " << key << " if no Layer set" << endl; return; }
 	vector_tile::Tile_Value v;
 	v.set_float_value(val);
-	outputs.back().second.push_back(attributeStore.store_key_value(key, v));
+	outputs.back().second.push_back(attributeStore.store_key_value(key, v, 0));
 	setVectorLayerMetadata(outputs.back().first->layer, key, 1);
 }
 
-void OsmLuaProcessing::AttributeBoolean(const string &key, const bool val) {
+void OsmLuaProcessing::AttributeBoolean(const string &key, const bool val) { AttributeBooleanWithMinZoom(key,val,0); }
+void OsmLuaProcessing::AttributeBooleanWithMinZoom(const string &key, const bool val, const char minzoom) {
 	if (outputs.size()==0) { cerr << "Can't add Attribute " << key << " if no Layer set" << endl; return; }
 	vector_tile::Tile_Value v;
 	v.set_bool_value(val);
-	outputs.back().second.push_back(attributeStore.store_key_value(key, v));
+	outputs.back().second.push_back(attributeStore.store_key_value(key, v, 0));
```

Comment:
remove (, 0) when adding default

Reply:
Argh, that should have been `, minzoom` anyway! Copy & paste error. Fixed.

The reason for having separate `Attribute` and `AttributeWithMinZoom` methods (rather than just one with default parameters) is to make the Kaguya Lua bindings work - it's not beautifully elegant but I think it's better than using KAGUYA_FUNCTION_OVERLOADS, which is the other way that Kaguya provides.